### PR TITLE
fix(config.js): add `APP__PROXY_DOMAIN` environment variable

### PR DESCRIPTION
**Pull request by stegano**
Hi @ZitRos,

thank you for making this useful library.

I tried to apply this library to our service, I made several improvements.

Please review and merge if possible.

Thanks 😃

---

### Fixes
- fix(Dockerfile): add environment variables (92500885c21a3005beb782b2f11fc98eedf6306a)
- fix(config.js): add `APP__PROXY_DOMAIN` environment variable (44a5a62cf54e3018479f422b5696b74b3353419e)

### Docs
- docs(readme.md): add `APP__PROXY_DOMAIN` environment variable description (7ac9f6eb1eccc83dc62629e846819b0a1f811252)

### CODE REVIEW

**Comment by ZitRos** [comment]
Thanks Yongwoo for your contribution! I left some comments, let me know what you think.

Also, we have a closed beta of [dataunlocker.com](https://dataunlocker.com/) running, let me know if you are interested in joining it.

**Comment by stegano** [comment]
> This should have a better description, like "The domain name used as a proxy for **analytics** scripts. When set, the traffic will be proxied via this domain. When not set, the current request domain is used as a proxy domain".
> 
> Before environment variables, it will now be useful to also add a comment explaining that the resulting proxy URL requested by the browser is expected to be `protocol://$APP__PROXY_DOMAIN$APP__STRIPPED_PATH/*(masked-url)*`.

Hi @ZitRos Thanks for your comments 😀
As your comments gave us, the `readme.md` documentation has been updated. Please tell me if you have any more corrections! (I'm not good at English, please understand 😅 , Could you please correct readme.md in my PR if possible?)

**Comment by stegano** [comment]
> Why do you need these environment variables in a Dockerfile? We need these environment variables for the runtime (i.e. `docker run ... -e APP__ENV_NAME=prod`), not during the build time.

If you use ARG and ENV at the same time as shown below, you can set the default environment variable value at build time and overwrite the environment variable set at build time at runtime.

Therefore, it can be applied in the order of Build time environment variable ⇛ Runtime environment variable ⇛ Default value in the source code.

```bash
ARG VARIABLE=(default)
ENV VARIABLE=$VARIABLE
```

If you think the above settings don't suit your purpose, or don't, please tell me. I'll fix it again (I respect your opinion)

**Comment by ZitRos** [comment]
Thank you!

I’ll check it tomorrow once again. Quick opinion on env variables in
Dockerfile: it’s just the first time I see env vars in Dockerfile used for
runtime (= it’s not a usual thing). Usually, they are set by the container
orchestration tool (i.e. when starting the container). The only
disadvantage I see is a need to re-declare all variables in the docker file
(hence, when introducing a new env variable or changing defaults you can
forgot to change it in the Dockerfile). Let me know what is an advantage of
putting env variables there — I’m curious! :)

Nikita

**Comment by stegano** [comment]
I thought the advantage of this is that you can flexibly modify the default values of environment variables without modifying the source code.

But when I think about it again, I think it would be better to use the build environment variable and execution environment variable information provided by Docker according to the purpose.

Thank you for making me think again. 😄
Please check it tomorrow and ask for additional opinions.

Thanks!

File: config.js [hunk 1]
```diff
@@ -12,7 +13,8 @@ const hostsWhitelistRegex = (() => {
 })();
 
 console.log("Environment variables:");
-console.log(`APP__STRIPPED_PATH=${strippedPath} (path added to original host in analytics scripts)`);
+console.log(`APP__PROXY_DOMAIN=${proxyDomain} (path added to original host replace proxy host`);
```
Comment (ZitRos) [inline config.js:16]: Comments get mixed; I'll fix this after the merge

**Comment by ZitRos** [comment]
Done! `docker pull zitros/analytics-saviour:1.1.0` is now live.

**Comment by stegano** [comment]
thanks!😀

**Pull request merged**